src/UserAction/StepExecutorTrait.php

```diff
@@ -102,18 +105,55 @@ protected function setFormField(Form $form, array $fields, string $step): Form
         foreach ($fields as $k => $val) {
             $form->getControl($k)->set($val);
         }
+
         $this->hook(self::HOOK_STEP, [$step, $form]);
 
         return $form;
     }
 
+    /**
+     * Set model for userAction arguments.
+     * Override existing argument with model definition.
+     */
+    protected function getModelForArgs(): Model
+    {
+        $this->cloneAction = clone $this->getAction();
```

Comment:
why do we need to clone, what is the purpose of using cloned&modified action internally?

this can lead also to many bugs when the action is semi-shallowly cloned, like when it has Closures bound still to the original action

Reply:
Cloning was to duplicate Action arguments when set by model only, i.e. each editable model field need to be added as an argument to the UserAction. However, I realized I do not need to clone UserAction but only it's arguments.

A fix is on this way.